spec/unit/shoulda/matchers/active_record/have_encrypt_attributes_matcher_spec.rb

```diff
@@ -0,0 +1,139 @@
+require 'unit_spec_helper'
+
+describe Shoulda::Matchers::ActiveRecord::HaveEncryptAttributeMatcher, type: :model do
+  if rails_version >= 7.0
+    context 'a encrypt attribute' do
+      it 'accepts' do
+        expect(with_encrypt_attr).to have_encrypt_attribute(:attr)
+      end
+    end
+
+    context 'an attribute that is not part of the encrypted-attributes set' do
+      it 'rejects being encrypted' do
+        model = define_model :example, attr: :string, other: :string do
+          encrypts :attr
+        end.new
+
+        expect(model).not_to have_encrypt_attribute(:other)
+      end
+    end
+
+    context 'an attribute on a class with no encrypt attributes' do
+      it 'rejects being encrypted' do
+        expect(define_model(:example, attr: :string).new).
+          not_to have_encrypt_attribute(:attr)
+      end
+
+      it 'assigns a failure message' do
+        model = define_model(:example, attr: :string).new
+        matcher = have_encrypt_attribute(:attr)
+
+        matcher.matches?(model)
+
+        expect(matcher.failure_message).to match(/Expect attr should be encrypted, but not/)
+      end
+    end
```

Comment:
We could improve all the specs that are checking for the `failure_message` directly to actually check if the matcher is failing with the correct message. That's better than checking directly the `failure_message`.

Here's an example, but you can also check the [spec/unit/shoulda/matchers/active_model/validate_comparison_of_matcher_spec.rb](https://github.com/thoughtbot/shoulda-matchers/blob/main/spec/unit/shoulda/matchers/active_model/validate_comparison_of_matcher_spec.rb) file for more examples.
```ruby
        context 'when the column is an string column' do
          it 'rejects and raise an error' do
            record = build_record_validating_comparison(column_type: :string, equal_to: 10)

            assertion = lambda do
              expect(record).to validate_comparison_of(:attr).is_equal_to(10)
            end

            expect(&assertion).to fail_with_message(<<~MESSAGE)
Expected Example to validate that :attr looks like a value equal to 10,
but this could not be proved.
  After setting :attr to ‹"10"›, the matcher expected the Example to be
  valid, but it was invalid instead, producing these validation errors:

  * attr: ["must be equal to 10"]
            MESSAGE
          end
        end
```